src/cosp.F90

```diff
@@ -951,12 +1542,53 @@ function COSP_SIMULATOR(cospIN,cospgridIN,cospOUT,start_idx,stop_idx,debug)
        endif
 
        ! Call simulator
-       call icarus_column(isccpIN%npoints, isccpIN%ncolumns,isccp_boxtau(:,:),           &
-                          isccp_boxptop(:,:)/100._wp, isccpIN%sunlit,isccp_boxttop,      &
-                          cospOUT%isccp_fq(ij:ik,:,:),                                   &
-                          cospOUT%isccp_meanalbedocld(ij:ik),                            &
-                          cospOUT%isccp_meanptop(ij:ik),cospOUT%isccp_meantaucld(ij:ik), &
-                          cospOUT%isccp_totalcldarea(ij:ik),cospOUT%isccp_meantb(ij:ik))
+       if (cospIN % cospswathsIN(1) % N_inst_swaths .gt. 0) then ! Trigger use of swathed arrays
+          if (isccpIN%Npoints .gt. 0) then
+             ! Operate at the masked format.
+             allocate(temp_isccp_fq(isccpIN%Npoints,numISCCPTauBins,numISCCPPresBins),   &
+                      temp_isccp_meanalbedocld(isccpIN%Npoints),                         &
+                      temp_isccp_meanptop(isccpIN%Npoints),                              &
+                      temp_isccp_meantaucld(isccpIN%Npoints),                            &
+                      temp_isccp_totalcldarea(isccpIN%Npoints),                          &
+                      temp_isccp_meantb(isccpIN%Npoints))
+
+             call icarus_column(isccpIN%npoints, isccpIN%ncolumns,isccp_boxtau(:,:),     &
+                                isccp_boxptop(:,:)/100._wp,isccpIN%sunlit,isccp_boxttop(:,:), &
+                                temp_isccp_fq,                                           &
```

Comment:
Are these temporaries needed. Or could we have the array reference in the argument list?
For example, replacing `temp_isccp_fq` with `cospOUT%isccp_fq(ij+int(ISCCP_MASK_INDICES)-1,:,:)`
If so, we could keep the same simulator calls for normal and swath modes.